source/s3_meta_request.c

```diff
@@ -1504,18 +1542,29 @@ void aws_s3_meta_request_send_request_finish_default(
 
         /* If the request failed due to an invalid (ie: unrecoverable) response status, or the meta request already
          * has a result, then make sure that this request isn't retried. */
-        if (error_code == AWS_ERROR_S3_INVALID_RESPONSE_STATUS ||
+        if (error_code == AWS_ERROR_S3_INVALID_RESPONSE_STATUS || error_code == AWS_ERROR_S3_GET_PART_SIZE_MISMATCH ||
             error_code == AWS_ERROR_S3_NON_RECOVERABLE_ASYNC_ERROR || meta_request_finishing) {
             finish_code = AWS_S3_CONNECTION_FINISH_CODE_FAILED;
-
-            AWS_LOGF_ERROR(
-                AWS_LS_S3_META_REQUEST,
-                "id=%p Meta request cannot recover from error %d (%s). (request=%p, response status=%d)",
-                (void *)meta_request,
-                error_code,
-                aws_error_str(error_code),
-                (void *)request,
-                response_status);
+            if (error_code == AWS_ERROR_S3_GET_PART_SIZE_MISMATCH) {
+                /* Log at info level instead of error as it's expected and not a fatal error */
+                AWS_LOGF_INFO(
+                    AWS_LS_S3_META_REQUEST,
+                    "id=%p Meta request cannot recover from error %d (%s). (request=%p, response status=%d)",
```

Comment:
do the error msg need to be updated? I'm confused how the comment says "not a fatal error" but the msg says "cannot recover from error"